test-framework/common/src/main/java/io/quarkus/test/common/NativeImageLauncher.java

```diff
@@ -60,6 +64,9 @@ public void start() throws IOException {
         args.add("-Dquarkus.http.port=" + port);
         args.add("-Dtest.url=" + TestHTTPResourceManager.getUri());
         args.add("-Dquarkus.log.file.path=" + PropertyTestUtil.getLogFileLocation());
+        if (profile != null) {
+            args.add("-Dquarkus.profile=" + profile);
```

Comment:
`prod` will be the default value here? I'm not totally sure.

Reply:
Yes, but there is an ongoing discussion about what should be the default.
(https://github.com/quarkusio/quarkus/issues/4371)

So, I've decided to not override the current default or next-to-be default. It's something we could revisit. 

So, this PR just offers the user the possibility to customize it.